packages/flutter/lib/src/material/back_button.dart

```diff
@@ -146,7 +156,11 @@ class CloseButton extends StatelessWidget {
       color: color,
       tooltip: MaterialLocalizations.of(context).closeButtonTooltip,
       onPressed: () {
-        Navigator.maybePop(context);
+        if (onPressed != null) {
+          onPressed();
```

Comment:
@HansMuller I'm wondering if the override callback should take in a single argument for `BuildContext` so that the callback has access to the same BuildContext as the original behavior. 

I know we don't do this elsewhere with other buttons like [FlatButton], but I'm curious on how we decide whether to expose it or not.

Reply:
`VoidCallback` is already powerful enough to allow access of different level `BuildContext`s. Explicit passing in `BuildContext` might be redundant.
See https://gist.github.com/imzyy/5332c6160b0d14cdc28e2834faa3bc6c and you can play with it in dartpad.